docs/source/gateway/index.rst

```diff
@@ -1,6 +1,1049 @@
+
+If you choose to include the API key directly, replace ``$OPENAI_API_KEY`` in the YAML file with your
+actual API key.
+
+If you prefer to use an environment variable (recommended), you can define it in your shell
+environment. For example:
+
+.. code-block:: bash
+
+     export OPENAI_API_KEY="your_openai_api_key"
+
+**Note:** Replace "your_openai_api_key" with your actual `OpenAI` API key.
+
+AI Gateway Configuration Details
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The MLflow AI Gateway service relies on a user-provided configuration file. It defines how the gateway interacts with various language model providers and dictates the routes that users can access.
+
+The configuration file is written in YAML and includes a series of sections, each representing a unique route. Each route section has a name, a type, and a model specification, which includes the provider, model name, and provider-specific configuration details.
+
+Here are the details of each configuration parameter:
+
+General Configuration Parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- **routes**: This is a list of route configurations. Each route represents a unique endpoint that maps to a particular language model service.
+
+Each route has the following configuration parameters:
+
+- **name**: This is the name of the route. It needs to be a unique name without spaces or any non-alphanumeric characters other than hyphen and underscore.
+
+- **route_type**: This specifies the type of service offered by this route. This determines the interface for inputs to a route and the returned outputs. Current supported route types are:
+
+  - "llm/v1/completions"
+  - "llm/v1/chat"
+  - "llm/v1/embeddings"
+
+- **model**: This defines the provider-specific details of the language model. It contains the following fields:
+
+  - **provider**: This indicates the provider of the AI model. It accepts the following values:
+
+    - "openai"
+    - "anthropic"
+    - "cohere"
+    - "azure" / "azuread"
+
+  - **name**: This is an optional field to specify the name of the model.
+  - **config**: This contains provider-specific configuration details.
+
+Provider-Specific Configuration Parameters
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+OpenAI
+++++++
+
+- **openai_api_key**: This is the API key for the OpenAI service. It is a required field.
+- **openai_api_type**: This is an optional field to specify the type of OpenAI API to use.
+- **openai_api_base**: This is the base URL for the OpenAI API. By default, it is set to "https://api.openai.com/v1".
+- **openai_api_version**: This is an optional field to specify the OpenAI API version.
+- **openai_organization**: This is an optional field to specify the organization in OpenAI.
+
+Cohere
+++++++
+
+- **api_key**: This is the API key for the Cohere service. It is a required field.
+- **api_base**: This is the base URL for the Cohere API. By default, it is set to "https://api.cohere.ai/v1".
```

Comment:
@BenWilson2 @harupy Is there a use case for configuring the Cohere / Anthropic API base? For OpenAI, I get it, since we have Azure. For Cohere / Anthropic, I think there's only one base? We aren't supporting base configuration on Databricks.

Unless I'm missing something, can we remove API base configuration for Cohere / Anthropic?

Reply:
@harupy removed these in https://github.com/mlflow/mlflow/pull/8967#pullrequestreview-1515677726. Can we remove them from the docs?